### DISCUSSION TITLE
Update only part of dynamic data?

### DISCUSSION THREAD

**Discussion by Anshi701** [vsg-dev/VulkanSceneGraph#1125] [Q&A] [answered]
Hello, let me explain the situation. I have a descriptor storage buffer, that have one vsg::Array assigned to it. Size of array is pretty large and updates very often.  If data is changed in that buffer, all memory is copying to GPU. And because updating is done very fast, FPS is decreasing.
Is there a way to make VSG update only part of this array? Maybe split one array to several ones smaller and make dirty() them separately?

**Comment by robertosfield** [comment, marked as answer]
There isn't presently an example that illustrates how to do it but it should be possible using proxy data objects that points to section of the large data.

vsg::Array<T>/Array2D<T>/Array3D<T> have support for using using another storage vsg::Data as the data storage and you can offset into this vsg::Data so address just a portion of it.   Then you can create a vsg::BufferInfo that use the portion of the vsg::Array.

These proxy data objects are lightweight as the don't store any data themselves so this extra wrapping really is just a mechanism for specifying the portion you want to work with.

I hope this make sense.

**Reply by Anshi701** [reply]
Thanks for reply! Yes, that does make sence, though right now I couldn't do it.
